### PR TITLE
feat: add support for the client acks of socket.io

### PR DESCRIPTION
**Pull request by dulichan**
@pleerock what do you think about adding support for client ack in socket.io. I did the above change in the handleSuccessResult function to enable this. But there is a small conflict in this working with emitOnSuccess functionality. Any ideas?

### CODE REVIEW

File: src/SocketControllerExecutor.ts [hunk 1]
```diff
@@ -216,14 +216,18 @@ export class SocketControllerExecutor {
...
             const transformOptions = action.emitOnSuccess.classTransformOptions || this.classToPlainTransformOptions;
             let transformedResult = this.useClassTransformer && result instanceof Object ? classToPlain(result, transformOptions) : result;
             socket.emit(action.emitOnSuccess.value, transformedResult);
 
         } else if ((result === null || result === undefined) && action.emitOnSuccess && !action.skipEmitOnEmptyResult) {
             socket.emit(action.emitOnSuccess.value);
+        } else if (result !== null && result !== undefined && clientCallback) {
+            clientCallback(result);
+        } else if ((result === null || result === undefined) && clientCallback) {
+            clientCallback("received");
```
Comment (pleerock) [inline src/SocketControllerExecutor.ts]: what does it mean "received"? Smells bad...

**Comment by pleerock** [comment]
why there is a conflict? Cant  `socket.emit(action.emitOnSuccess.value);` and new callback live together? If no, then why?

**Comment by dulichan** [comment]
@pleerock there isn't a conflict. Do you see a code conflict? I

File: src/SocketControllerExecutor.ts (hunk 1 above)
Comment (dulichan) [inline src/SocketControllerExecutor.ts]: Received is for situations where the function returns nothing but the client is expecting a callback. Any good ideas for this?

Comment (pleerock) [inline src/SocketControllerExecutor.ts]: I don't know can we simply call `clientCallback()` without arguments?

Comment (dulichan) [inline src/SocketControllerExecutor.ts]: That's a good idea

**Comment by rustamwin** [comment]
#6

**Comment by rustamwin** [comment]
This package does not know that the 2nd argument is a function, client can send few arguments

**Comment by infnada** [comment]
Any news on this, it doesn't seems too hard to implement like #6 Thanks!

**Comment by NoNameProvided** [comment]
Closing as no response from the author.

**Pull request closed without merging**